src/Experiment/src/AssemblyBuilder.cs

```diff
@@ -1,27 +1,73 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-﻿using System;
 using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-using System.Reflection;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Reflection.PortableExecutable;
 
 namespace System.Reflection.Emit.Experimental
 {
+
     public class AssemblyBuilder: System.Reflection.Assembly
     {
-        internal AssemblyBuilder() { }
+        private bool _previouslySaved=false;
+        private AssemblyName _assemblyName;
+        private MetadataBuilder _metadata;
+        private ModuleBuilder? _module;
+
+        private AssemblyBuilder(AssemblyName name) 
+        {
+            _assemblyName = name;
+            _metadata = new MetadataBuilder();
+        }
 
         public void Save(string assemblyFileName)
         {
-            //New method
+            if(_previouslySaved) // You cannot save an assembly multiple times. This is consistent with Save() in .Net Framework.
+            {
+                throw new InvalidOperationException("Cannot save an assembly multiple times");
+            }
+
+            if (assemblyFileName == null)
+            {
+                throw new ArgumentNullException(nameof(assemblyFileName));
+            }
+
+            if(_module == null)
```

Comment:
```suggestion
            if (_assemblyName == null || _assemblyName.Name == null)
            {
                throw new ArgumentNullException(nameof(_assemblyName));
            }
            
            if(_module == null)
```
Ok, this was actually needed to make the compiler happy. Adding it back